### PR TITLE
add the traffic_forward:contains two diversion modes: DNAT and DNS

### PR DESCRIPTION
**Pull request by JcJinChen**
#### What this PR does / why we need it:
DNAT and DNS, as the two diversion modes of yurt-tunnel, have the same function, so I put iptables and dns in the  directory:pkg/yurttunnel/traffic_forward, and I will solve this [issue548](https://github.com/openyurtio/openyurt/issues/548) in the next PR.

#### Special notes for your reviewer:
@rambohe-ch 

#### Does this PR introduce a user-facing change?

NONE

### CODE REVIEW

**Comment by rambohe-ch** [comment]
@JcJinChen please fix the golangci lint error.

**Comment by rambohe-ch** [comment]
@JcJinChen how about modify `traffic_forward` to `trafficforward`?  because all directory names have no `_`.

**Comment by JcJinChen** [comment]
> @JcJinChen how about modify `traffic_forward` to `trafficforward`? because all directory names have no `_`.

get it.

File: cmd/yurt-tunnel-server/app/start.go [hunk 1]
```diff
@@ -18,17 +18,19 @@ package app
...
 	"sync"
 	"time"
 
 	"github.com/openyurtio/openyurt/cmd/yurt-tunnel-server/app/config"
 	"github.com/openyurtio/openyurt/cmd/yurt-tunnel-server/app/options"
 	"github.com/openyurtio/openyurt/pkg/projectinfo"
-	"github.com/openyurtio/openyurt/pkg/yurttunnel/dns"
 	"github.com/openyurtio/openyurt/pkg/yurttunnel/handlerwrapper/initializer"
 	"github.com/openyurtio/openyurt/pkg/yurttunnel/handlerwrapper/wraphandler"
+
```
Comment (rambohe-ch) [inline cmd/yurt-tunnel-server/app/start.go:31]: delete the empty line

File: cmd/yurt-tunnel-server/app/start.go [hunk 2]
```diff
@@ -18,17 +18,19 @@ package app
 
 import (
 	"fmt"
+	"github.com/openyurtio/openyurt/pkg/yurttunnel/trafficforward/dns"
+	"github.com/openyurtio/openyurt/pkg/yurttunnel/trafficforward/iptables"
```
Comment (rambohe-ch) [inline cmd/yurt-tunnel-server/app/start.go:22]: move these two packages to align with `github.com/openyurtio/openyurt`

File: pkg/yurtctl/cmd/revert/revert.go [hunk 3]
```diff
@@ -19,6 +19,7 @@ package revert
 import (
 	"context"
 	"fmt"
+	tunneldns "github.com/openyurtio/openyurt/pkg/yurttunnel/trafficforward/dns"
```
Comment (rambohe-ch) [inline pkg/yurtctl/cmd/revert/revert.go:22]: the same as above comment

**Comment by rambohe-ch** [comment]
@JcJinChen Would you be able to explain that `go.sum` file updated?

File: cmd/yurt-tunnel-server/app/start.go (hunk 2 above)
Comment (JcJinChen) [reply cmd/yurt-tunnel-server/app/start.go:22]: ok, I know. Thank you for advices.

**Pull request closed without merging**